feat!: upgrade to snakemake v9

nh13 opened the pull request:
BREAKING CHANGE: Now requires snakemake ^9.0.0 (previously ^8.27.1)

- Refactor SnakemakeLogger to use Python's standard logging.Handler instead of the removed log_handlers parameter in OutputSettings
- Add handler directly to snakemake's logger_manager after API setup
- Update RuleLog.many_from_log_file to deduplicate errors by jobid (snakemake v9 logs errors twice: inline and in summary)
- Extend Python version support to include 3.13

We could use the plugin system if we have a bit more information, see: https://github.com/snakemake/snakemake/issues/3886.

nh13 committed:
feat!: upgrade to snakemake v9

BREAKING CHANGE: Now requires snakemake ^9.0.0 (previously ^8.27.1)

- Refactor SnakemakeLogger to use Python's standard logging.Handler
  instead of the removed log_handlers parameter in OutputSettings
- Add handler directly to snakemake's logger_manager after API setup
- Update RuleLog.many_from_log_file to deduplicate errors by jobid
  (snakemake v9 logs errors twice: inline and in summary)
- Extend Python version support to include 3.13

clintval commented on fgsmk/log.py:
```diff
@@ -62,6 +71,21 @@ def many_from_log_file(cls, log_dir: Path, snakemake_log: Path) -> list["RuleLog
+
+            # Extract jobid for deduplication
+            jobid: str | None = None
+            for line in lines[1:5]:  # Check next few lines for jobid
```
Why 5? Is there a chance we'll miss the log statement we're searching for?

clintval commented on fgsmk/log.py:
```diff
@@ -62,6 +71,21 @@ def many_from_log_file(cls, log_dir: Path, snakemake_log: Path) -> list["RuleLog
+            jobid: str | None = None
+            for line in lines[1:5]:  # Check next few lines for jobid
+                if line.startswith(cls.JOBID_PREFIX):
+                    jobid = line[len(cls.JOBID_PREFIX) :].strip()
```
Why not use [`str.lstrip()`](https://docs.python.org/3/library/stdtypes.html#str.lstrip)?

nh13 committed:
refactor: address PR review comments in log.py

- Add RuleName and JobId type aliases for clarity
- Use removeprefix() instead of manual slicing for jobid extraction
- Remove arbitrary lines[1:5] limit; search until LOG_PREFIX/SHELL_PREFIX
